src/librbd/api/Mirror.cc

```diff
@@ -1271,6 +1271,47 @@ int Mirror<I>::mode_set(librados::IoCtx& io_ctx,
   return 0;
 }
 
+template <typename I>
+int Mirror<I>::remote_namespace_get(librados::IoCtx& io_ctx,
+                                    std::string* remote_namespace) {
+
+  CephContext *cct = reinterpret_cast<CephContext *>(io_ctx.cct());
+  ldout(cct, 20) << dendl;
+
+  int r = cls_client::mirror_remote_namespace_get(&io_ctx, remote_namespace);
+  if (r < 0 && r != -ENOENT) {
```

Comment:
I am changing this to the following:
In case of ENOENT or ENOPNOTSUPP,  set the value to that of io_ctx.get_namespace().

For the "remote namespace is set to a default namespace" case, the key would still exist but with an empty value.

Reply:
Conversation moved to https://github.com/ceph/ceph/pull/59417/files#r1755088594